types/mui-datatables/index.d.ts

```diff
@@ -311,7 +325,7 @@ export type MUIDataTableOptions = Partial<{
         data: Array<{ index: number; dataIndex: number }>;
     }) => void;
     onRowSelectionChange: (currentRowsSelected: any[], allRowsSelected: any[], rowsSelected?: any[]) => void;
-    onSearchChange: (searchText: string) => void;
+    onSearchChange: (searchText: string | null) => void;
```

Comment:
[`searchText` is null](https://github.com/gregnb/mui-datatables/blob/d50ff23334c28035a244233b844501f89afddf7e/src/MUIDataTable.js#L1326) when the search field is closed (`this.state.searchText` becomes null)